lottery/io/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+#![no_std]
+
+use codec::{Decode, Encode};
+use gstd::{prelude::*, ActorId};
+use scale_info::TypeInfo;
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub struct FtInitConfig {
+    pub name: String,
+    pub symbol: String,
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub struct LotteryState {
+    pub lottery_started: bool,
+    pub lottery_start_time: u64,
+    pub lottery_duration: u64,
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone, Copy)]
+pub struct Player {
+    pub player_id: ActorId,
+    pub balance: u128,
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub enum Action {
+    Enter(u128),
+    StartLottery {
+        duration: u64,
+        token_address: Option<ActorId>,
+    },
+    LotteryState,
+    PickWinner,
+    BalanceOf(u32),
+    GetPlayers,
+    LeaveLottery(u32),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    LotteryState(LotteryState),
+    Winner(u32),
+    Balance(u128),
+    Players(BTreeMap<u32, Player>),
+    PlayerAdded(u32),
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub enum FtAction {
```

Comment:
There is no need to define fungible token actions inside your contract, you can import them from fungible-token contract.